### ISSUE TITLE
Question: about creating indexes for a model

### ISSUE THREAD

**Issue by shopapps** [Klaasie/laravel-scout-solr-engine#8]
Firstly, Great work on this package.  Just needs updating to allow for install with laravel 10 & scout 10 when you get a moment ;-)

I was wondering if you are able to expand on your notes about the creation of Cores (indexes).  If I try:

```
php artisan scout:index users_index --key=id
```

then I get an error back:

```
Solr HTTP error: OK (400)
{
  "responseHeader":{
    "status":400,
    "QTime":40},
  "error":{
    "metadata":[
      "error-class","org.apache.solr.common.SolrException",
      "root-error-class","org.apache.solr.common.SolrException"],
    "msg":"Error CREATEing SolrCore 'users_index': coreNodeName missing {configSet=_default}",
    "code":400}}
```

this was with a vanilla call to a local "cloud" solr

```
bin/solr start -e cloud -noprompt
```

(this is using solr 9.2.1 btw :-) ) is there a recommended way to start solr or a setting needed to allow it to work?

I can see there is a **_default** in a configsets folder in the solr folder structure.

I need to read up more on solr as I still get confused between **Core**  & **Collection** - lol

also..  if I try:

```
php artisan scout:import "App\Models\User"
```

I see an error, which I assume is due to the index not being created.  because If I manually create the collection in the 

```
http://127.0.0.1:8983/solr/#/~collections
```

menu, then the scout:import works. :-) 

```
   Solarium\Exception\HttpException 

  Solr HTTP error: OK (404)
<p>
  Searching for Solr?<br/>
  You must type the correct path.<br/>
  Solr will respond.
</p>
```

## Screenshot of the solr folders showing the **_default** folder.
 
<img width="817" alt="Screenshot 2023-05-11 at 08 38 21" src="https://github.com/Klaasie/laravel-scout-solr-engine/assets/8806928/12a633fa-ac00-41b1-b0af-e8d148464863">

**Comment by Klaasie** [comment]
Hi! 

Sorry for late response to this question. I'll have to modify my notification settings for this repository.. 😅 

I've updated the package to support Laravel and Scout 10.

As for your question:
Solr requires a configset when you use the `scout:index` command. This way SOLR knows which default settings to apply to the new core. More documentation on this is provided by SOLR here: https://solr.apache.org/guide/solr/latest/configuration-guide/config-sets.html
This is why, in the example setup, I have docker copy the default core configuration supplied by SOLR to the data folder: https://github.com/Klaasie/laravel-scout-solr-engine/blob/master/example/docker/solr/Dockerfile#L5
When this folder is present the `scout:index` command should work. Additionally, if you have your own custom "default" core settings, you can provide the name through the config file: https://github.com/Klaasie/laravel-scout-solr-engine/blob/master/config/scout-solr.php#L17

I hope this answer is any help to you. If you have additional questions feel free to ask and I'll make sure to respond in a timely manner.

**Comment by shopapps** [comment]
hi, thanks for responding.

makes sense, so in your case the solr server is running on the same server hence you have access to those files?

I'll get my ops guys to run that command on each of the clusters.  For the benefit of others you should do something like this: 

```bash
sudo cp -r /opt/solr-9.2.1/server/solr/configsets /var/solr/data
sudo chown -R solr:solr /var/solr/data/configsets
```

then checking that folder you should see something like:

```bash
sudo ls -lah /var/solr/data/configsets/

total 16K
drwxr-xr-x  4 solr solr 4.0K Jul 13 07:03 .
drwxr-x--- 13 solr solr 4.0K Jul 13 07:03 ..
drwxr-xr-x  3 solr solr 4.0K Jul 13 07:03 _default
drwxr-xr-x  3 solr solr 4.0K Jul 13 07:03 sample_techproducts_configs

```
most likely you can safely delete the `sample_techproducts_configs` folder from there unless you are using it :-)

thanks again Klassie

**Comment by Klaasie** [comment]
Hi!

I'm not sure if I'm understanding this correctly, but in the application provided in the example folder Solr is running in a docker container that can be accessed. So in that container, the configsets needs to be located in the `/var/solr/data` folder in order for Solr to be able to use them.

However, I'm now reading the documentation on configsets in SolrCloud clusters and in this case the `_default` should already be available and is used as a fallback:
> When you create a collection in SolrCloud, you can specify a named configset. If you don’t, then the _default will be copied and given a unique name for use by the new collection.

Perhaps the error is being caused because the code explicitly sets the configset to `_default` rather than leaving this blank. 
In that case omitting [this line](https://github.com/Klaasie/laravel-scout-solr-engine/blob/master/src/Engines/SolrEngine.php#L179) might fix the issue. 

I'll do some testing on this on the weekend.

**Comment by Klaasie** [comment]
I've looked into the compatibility of this package with Solr Cloud and ran into the same issues. omitting the `_default` config set didn't work so I dug some more into the documentation. Turns out Solr recommends to use the Collection API for Solr Cloud instead of the Core Admin API.

I've created an PR providing a `cloud` config key and implementation through the Collection API in case Solr Cloud is used. You can find it here: https://github.com/Klaasie/laravel-scout-solr-engine/pull/11

I'll give this PR some more testing but so far I was able to both create and delete collections without having to copy and files or folders into `/var/solr/data`. Also the import command works as expected.
It's useful to know that if the `_default` config set doesn't satisfy you needs, that with Solr Cloud you'd need to upload your own config set through ZooKeeper. More on this topic can be found here: https://solr.apache.org/guide/solr/latest/configuration-guide/config-sets.html#configsets-in-solrcloud-clusters

Additionally I moved the `config_set` config value from the root to the endpoint cores definition since I felt it makes sense that one might want to use different config sets for different models.

**Comment by shopapps** [comment]
looks good...  I will give it a test on Monday as well... 🤞

**Comment by Klaasie** [comment]
It's been a while since we worked on this issue. The changes have been merged into master and are part of the V3.0.0 release.
